cypress/e2e/12-bean_support/bean_support.cy.js

```diff
@@ -0,0 +1,141 @@
+    cy.checkCodeSpanLine('- beans:');
+    cy.checkCodeSpanLine('- name: test');
+    cy.checkCodeSpanLine('type: org.acme');
+    cy.checkCodeSpanLine('properties:');
+    cy.checkCodeSpanLine('property1: value1');
+  });
+
+  it('User inserts new object property using the bean editor modal', () => {
+    cy.uploadFixture('Beans.yaml');
+
+    cy.get('[data-testid=toolbar-metadata-beans-btn]').click();
+    cy.get('[data-testid="metadata-row-0"]').click();
+
+    cy.get('[data-testid="properties-add-object-property--btn"]').click();
+
+    cy.get('[data-testid="properties--placeholder-name-input"]').click();
+    cy.get('[data-testid="properties--placeholder-name-input"]').type('object');
+
+    // Click outside the "name" input, to display the object property handling options
+    cy.get('.metadataEditorModal').click(0, 0);
+
+    cy.get('[data-testid="properties-add-string-property-object-btn"]').click();
+    cy.get('[data-testid="properties-object-placeholder-name-input"]').click();
+    cy.get('[data-testid="properties-object-placeholder-name-input"]').type('object-prop');
+
+    // Click outside the "name" input, to display the object property value input
+    cy.get('.metadataEditorModal').click(0, 0);
+
+    cy.get('[data-testid=properties-object-object-prop-value-input]').click();
+    cy.get('[data-testid=properties-object-object-prop-value-input]').clear().type('newvalue');
+
+    cy.get('[data-ouia-component-id="OUIA-Generated-Modal-large-2-ModalBoxCloseButton"]').click();
+
+    // CHECK the bean was reflected in the code editor
+    cy.checkCodeSpanLine('object:');
+    cy.checkCodeSpanLine('object-prop: newvalue');
+  });
+
+  it.skip('User edits a bean using the bean editor modal', () => {
+    cy.uploadFixture('Beans.yaml');
+
+    cy.get('[data-testid=toolbar-metadata-beans-btn]').click();
+    cy.get('[data-testid="metadata-row-0"]').click();
+
+    cy.get(`input[name="name"]`).clear().type('rename');
+    cy.get(`input[name="type"]`).clear().type('com.emca');
+
+    cy.get('.metadataEditorModal').click(0, 0);
+
+    cy.get('[data-testid="metadata-row-0"]').as('row');
+    cy.get('@row').find('td').eq(0).should('contain', 'rename');
+    cy.get('@row').find('td').eq(1).should('contain', 'com.emca');
+
+    cy.get('[data-testid=properties-add-string-property--btn]').eq(0).click();
+    cy.get('[data-testid=properties-property1-name-input]').click();
+    cy.get('[data-testid=properties-property1-name-input]').clear().type('newtest');
+
+    // Click outside the "name" input, to display the "value" input
+    cy.get('.metadataEditorModal').click(0, 0);
+    cy.get('[data-testid=properties-newtest-value-input]').click();
+    cy.get('[data-testid=properties-newtest-value-input]').clear().type('newvalue');
+
+    // CURRENTLY BLOCKED BY - https://github.com/KaotoIO/kaoto-ui/issues/2168
+    cy.get('[data-ouia-component-id="OUIA-Generated-Modal-large-2-ModalBoxCloseButton"]').click();
```

Comment:
Note for future ourselves: Update the OUIA selector to use something that doesn't change depending on the rendering order.